src/wfs/src/test/java/org/geoserver/wfs/GetFeatureCachingTest.java

```diff
@@ -6,21 +6,23 @@
 
 import org.geoserver.wfs.v2_0.GetFeatureTest;
 import org.junit.AfterClass;
+import org.junit.Assert;
 import org.junit.BeforeClass;
 
 /**
  * Test making sure GetFeature still works when the caching machinery is off (since most test
  * datasets will actually be cached in memory)
  */
-public class GetFeatureNoCachingTest extends GetFeatureTest {
+public class GetFeatureCachingTest extends GetFeatureTest {
 
     @BeforeClass
-    public static void disableCaching() {
-        FeatureSizeFeatureCollection.setFeatureCacheLimit(0);
+    public static void enableCaching() {
+        Assert.assertEquals(0, FeatureSizeFeatureCollection.DEFAULT_CACHE_SIZE);
+        FeatureSizeFeatureCollection.setFeatureCacheLimit(16);
     }
 
     @AfterClass
-    public static void enableCaching() {
+    public static void disableCaching() {
         FeatureSizeFeatureCollection.setFeatureCacheLimit(
```

Comment:
While this is a programmatic way to enable, GeoServer often uses a system property (to control internal functionality changes)

Reply:
I believe this controlled by a system property: https://github.com/ccri/geoserver/blob/a850eba6036d39255f74c82995dd89f5d07ff215/src/wfs/src/main/java/org/geoserver/wfs/FeatureSizeFeatureCollection.java#L55-L57.

Reply:
I think this method was just used for testing - the initial value is controlled by the system property here: https://github.com/ccri/geoserver/blob/a850eba6036d39255f74c82995dd89f5d07ff215/src/wfs/src/main/java/org/geoserver/wfs/FeatureSizeFeatureCollection.java#L55-L57

Reply:
Jinx!  I said it first!

Reply:
Do you know if "org.geoserver.wfs.getfeature.cachelimit" is documented somewhere?

However since this a back port I do not wish to hold it up on docs.

Reply:
it does not appear to be documented anywhere, but I don't mind adding it, I'm just not sure where would be appropriate.